util/conditions/v1beta2/summary_test.go

```diff
@@ -0,0 +1,224 @@
+			},
+		},
+		{
+			name: "One unknown (no issues)",
+			conditions: []metav1.Condition{
+				{Type: "B", Status: metav1.ConditionTrue, Reason: "Reason-B", Message: "Message-B"},       // info
+				{Type: "A", Status: metav1.ConditionTrue, Reason: "Reason-A", Message: "Message-A"},       // info
+				{Type: "!C", Status: metav1.ConditionUnknown, Reason: "Reason-!C", Message: "Message-!C"}, // unknown
+			},
+			conditionType: AvailableCondition,
+			options:       []SummaryOption{ForConditionTypes{"A", "B", "!C"}, WithNegativePolarityConditionTypes{"!C"}},
+			want: &metav1.Condition{
+				Type:    AvailableCondition,
+				Status:  metav1.ConditionUnknown,    // Unknown because there is one unknown
+				Reason:  "Reason-!C",                // Picking the reason from the only existing unknown
+				Message: "!C (Unknown): Message-!C", // messages from all the issues & unknown conditions (info dropped)
+			},
+		},
+		{
+			name: "More than one unknown (no issues)",
+			conditions: []metav1.Condition{
+				{Type: "B", Status: metav1.ConditionUnknown, Reason: "Reason-B", Message: "Message-B"},    // unknown
+				{Type: "A", Status: metav1.ConditionTrue, Reason: "Reason-A", Message: "Message-A"},       // info
+				{Type: "!C", Status: metav1.ConditionUnknown, Reason: "Reason-!C", Message: "Message-!C"}, // unknown
+			},
+			conditionType: AvailableCondition,
+			options:       []SummaryOption{ForConditionTypes{"A", "B", "!C"}, WithNegativePolarityConditionTypes{"!C"}},
+			want: &metav1.Condition{
+				Type:    AvailableCondition,
+				Status:  metav1.ConditionUnknown,                            // Unknown because there are many unknown
+				Reason:  MultipleUnknownReported,                            // Using a generic reason
+				Message: "B (Unknown): Message-B; !C (Unknown): Message-!C", // messages from all the issues & unknown conditions (info dropped)
+			},
+		},
+
+		{
+			name: "More than one info (no issues, no unknown)",
+			conditions: []metav1.Condition{
+				{Type: "B", Status: metav1.ConditionTrue, Reason: "Reason-B", Message: "Message-B"},     // info
+				{Type: "A", Status: metav1.ConditionTrue, Reason: "Reason-A", Message: ""},              // info
+				{Type: "!C", Status: metav1.ConditionFalse, Reason: "Reason-!C", Message: "Message-!C"}, // info
+			},
+			conditionType: AvailableCondition,
+			options:       []SummaryOption{ForConditionTypes{"A", "B", "!C"}, WithNegativePolarityConditionTypes{"!C"}, WithMergeStrategy{newDefaultMergeStrategy()}},
+			want: &metav1.Condition{
+				Type:    AvailableCondition,
+				Status:  metav1.ConditionTrue,                          // True because there are many info
+				Reason:  MultipleInfoReason,                            // Using a generic reason
+				Message: "B (True): Message-B; !C (False): Message-!C", // messages from all the info conditions (empty messages are dropped)
```

Comment:
Would it make sense to include A (True)?

Reply:
Discussed, let's drop the Status (True/False) from summary messages

Reply:
Done, PTAL